src/lib/CONFIG/config.cpp

```diff
@@ -61,61 +117,175 @@ void TxConfig::Load()
     ESP_ERROR_CHECK( err );
     ESP_ERROR_CHECK(nvs_open("ELRS", NVS_READWRITE, &handle));
 
-    // read version field
-    if(nvs_get_u32(handle, "tx_version", &m_config.version) != ESP_ERR_NVS_NOT_FOUND
-        && m_config.version == (TX_CONFIG_VERSION | TX_CONFIG_MAGIC))
+    // Try to load the version and make sure it is a TX config
+    uint32_t version = 0;
+    if (nvs_get_u32(handle, "tx_version", &version) == ESP_OK && ((version & CONFIG_MAGIC_MASK) == TX_CONFIG_MAGIC))
+        version = version & ~CONFIG_MAGIC_MASK;
+    DBGLN("Config version %u", version);
+
+    // Can upgrade from any version 5 to current
+    if (version < 5)
     {
-        DBGLN("Found version %u config", TX_CONFIG_VERSION);
-        uint32_t value;
-        nvs_get_u32(handle, "vtx", &value);
-        m_config.vtxBand = value >> 24;
-        m_config.vtxChannel = value >> 16;
-        m_config.vtxPower = value >> 8;
-        m_config.vtxPitmode = value;
+        SetDefaults(true);
+        return;
+    }
 
-        nvs_get_u32(handle, "fan", &value);
-        m_config.fanMode = value;
+    SetDefaults(false);
 
-        nvs_get_u32(handle, "motion", &value);
-        m_config.motionMode = value;
+    uint32_t value;
+    uint8_t value8;
+    // vtx (v5)
+    value = 0;
+    nvs_get_u32(handle, "vtx", &value);
+    m_config.vtxBand = value >> 24;
+    m_config.vtxChannel = value >> 16;
+    m_config.vtxPower = value >> 8;
+    m_config.vtxPitmode = value;
 
-        uint8_t value8;
-        nvs_get_u8(handle, "fanthresh", &value8);
-        m_config.powerFanThreshold = value8;
+    // fanthresh (v5)
+    value8 = PWR_250mW;
+    nvs_get_u8(handle, "fanthresh", &value8);
+    m_config.powerFanThreshold = value8;
 
+    // Both of these were added to config v5 without incrementing the version
+    if (nvs_get_u32(handle, "fan", &value) == ESP_OK)
+        m_config.fanMode = value;
+    if (nvs_get_u32(handle, "motion", &value) == ESP_OK)
+        m_config.motionMode = value;
+
+    if (version >= 6)
+    {
+        // dvr (v6)
+        value8 = 0;
         nvs_get_u8(handle, "dvraux", &value8);
         m_config.dvrAux = value8;
+        value8 = 0;
         nvs_get_u8(handle, "dvrstartdelay", &value8);
         m_config.dvrStartDelay = value8;
+        value8 = 0;
         nvs_get_u8(handle, "dvrstopdelay", &value8);
         m_config.dvrStopDelay = value8;
```

Comment:
I added value=0 before each call otherwise the latter would get the value from the previous call if it as not set. Probably not required, but better safe than sorry.